app/controllers/course/experience_points_records_controller.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+class Course::ExperiencePointsRecordsController < Course::ComponentController
+  load_resource :course_user, through: :course
+  load_and_authorize_resource :experience_points_record, through: :course_user,
+                                                         class: Course::ExperiencePointsRecord.name
+
+  def index # :nodoc:
+    @experience_points_records =
```

Comment:
Yes, there's no need to use references. includes alone should be sufficient. references forces AR to generate a LEFT JOIN, which increases preload time.